[ECE] Doc clarification - add swap description (#7196)
elastic/docs-content: deploy-manage/deploy/cloud-enterprise/ece-manage-capacity.md

* [CPU quotas](#ece-alloc-cpu)
* [Processors setting](#ece-alloc-processors-setting)
* [Storage](#ece-alloc-storage)

> * [Swap consideration](#ece-alloc-swap-consideration)

Reviewer: This item is now out of place.

Author: Thanks, updated the doc to remove that isolated one.

::::{important}
Swap should be treated as an emergency safety net only — not as a way to overcommit memory or reduce host RAM. If a container runtime process (Docker or Podman) runs on swap, it can cause allocator failures due to API timeouts (visible as errors in `allocator.log`). Always ensure allocators are not over-allocated so the OS does not routinely rely on swap.
::::

### Swap sizing [ece-alloc-swap-sizing]

There is no fixed scalability recommendation for swap sizing, but 4 GB of swap per 32 GB of RAM has proven to be a reasonable safeguard for most ECE deployments. The minimum recommended swap space on ECE hosts is 512 MB.

Set `vm.swappiness=1` so that swap is used only as a last resort.

### Enabling swap [ece-alloc-enabling-swap]

Follow the host configuration steps for your OS to enable swap:

* [Ubuntu](https://www.elastic.co/docs/deploy-manage/deploy/cloud-enterprise/configure-host-ubuntu)
* [RHEL](https://www.elastic.co/docs/deploy-manage/deploy/cloud-enterprise/configure-host-rhel)

> * [SUSE](https://www.elastic.co/docs/deploy-manage/deploy/cloud-enterprise/configure-host-suse)

Reviewer: ```suggestion
* [Ubuntu](/deploy-manage/deploy/cloud-enterprise/configure-host-ubuntu.md)
* [RHEL](/deploy-manage/deploy/cloud-enterprise/configure-host-rhel.md)
* [SUSE](/deploy-manage/deploy/cloud-enterprise/configure-host-suse.md)
```

Fixing the links. But one extra question:

Do we have specific swap configuration instructions on those documents? I believe we don't, and probably it's not our thing to tell users the exact steps to configure swap memory at OS level, just telling them what amount of swap we recommend.

We could tell them that by default all OS enable swap by default, but if they haven't enabled swap they should follow the applicable guide to their operating system.

In our Ubuntu / RHEL and SuSE docs we don't have steps to enable swap, our steps about swap are kernel and grub loader considerations about swap (which is what really matters to ECE).

Author: Thanks @eedugon you are right, I should mention that configuring swap is not in the scope of elastic support.
I added an additional wording to make it clear:

> Furthermore, how you provision the swap space itself depends on your operating system and infrastructure provider — consult your OS or cloud provider's documentation for creating a swap file or partition. This is a topic out of Elastic support scope.

The override only persists during the lifecycle of the instance container. If a new container is created, for example during a `grow_and_shrink` plan or a vacate operation, the quota is reset to its default. To increase the storage ratio in a persistent way, [edit the instance configurations](ece-configuring-ece-instance-configurations-edit.md).
::::

> ## Swap consideration [ece-alloc-swap-consideration]

Reviewer: What if we move all `## Swap consideration` section as a sub-section of `## Memory`?

So:
## Memory
### Example
### Swap considerations

Doing that the swap part would feel better fit and allocated if we decide to keep it in this doc, more than a separate section.

And we should aim to make this section the shorter the better, just share the main ideas and considerations we have to share, without too much prose.

Author: Sounds great! 
Thank you Edu! 
Could you help me reorg the contents please?
Happy to do that too if you prefer.

> ## Swap consideration [ece-alloc-swap-consideration]

Reviewer: ```suggestion
## Swap memory considerations [ece-alloc-swap-consideration]
```

I think the comments about swap could be considered considerations more than a single consideration.

- explanation of what happens when swap is not enabled
- small sizing paragraph
- reminder to set swappiness at kernel level.
- statement to tell users to find their OS guides for swap configuration

While {{es}} nodes generally run with [swap disabled](../self-managed/setup-configuration-memory.md), ECE hosts should have swap enabled for stability reasons.

If an ECE host runs out of memory, the Linux out of memory (OOM) killer stops a random process on the runner. Having swap space available can prevent this from happening and protect the availability of ECE services.

::::{important}
Swap should be treated as an emergency safety net only — not as a way to overcommit memory or reduce host RAM. If a container runtime process (Docker or Podman) runs on swap, it can cause allocator failures due to API timeouts (visible as errors in `allocator.log`). Always ensure allocators are not over-allocated so the OS does not routinely rely on swap.
::::

> ### Swap sizing [ece-alloc-swap-sizing]

Reviewer: I wouldn't create a sizing sub-section for a single paragraph, more over if our sizing guidance is that we don't provide any special guidance. I think leaving it embedded in the section would be enough.

Author: I think we already did this? Thanks!

> ### Swap sizing [ece-alloc-swap-sizing]

Reviewer: ```suggestion
```

> There is no fixed scalability recommendation for swap sizing, but 4 GB of swap per 32 GB of RAM has proven to be a reasonable safeguard for most ECE deployments. The minimum recommended swap space on ECE hosts is 512 MB.

Reviewer: ```suggestion
There is no specific recommendation for sizing swap, but 4 GB of swap per 32 GB of RAM has proven to be a reasonable safeguard for most ECE installations. As a baseline, ECE hosts should have at least 512 MB of swap space.
```

For your consideration, reasons for the suggestion:
- fixed scalability recommendation sounds a bit odd
- As you know, an `ECE deployment`, or a `deployment` is typically used for an elasticsearch cluster, and we want our readers to have that mental link. Here we mean an ECE environment, so better to say ECE installation, environmnet or system, more than deployment.
- At the end the `minimum recommended` felt a bit weird when the sentence started with there's no fixed recommendation.

Author: Thank you!!

> Set `vm.swappiness=1` so that swap is used only as a last resort.

Reviewer: ```suggestion
Set the `vm.swappiness` kernel setting to `1`, as described in the [Configure your OS](/deploy-manage/deploy/cloud-enterprise/configure-operating-system.md) preparation guides, so that swap is used only as a last resort.
```

The current text had 2 issues:
- A reader might not understand what's that setting about.
- A reader might have no clue about how to do that (it's in our OS guide).

Reviewer: And maybe this paragraph could be moved at the end of the section, being the closure (after the message saying the the instructions to enable swap depend on the OS)

Author: Makes sense! Thanks!

Furthermore, how you provision the swap space itself depends on your operating system and infrastructure provider —

> consult your OS or cloud provider's documentation for creating a swap file or partition. This is a topic out of Elastic support scope.

Reviewer: ```suggestion
The method for provisioning swap space depends on your operating system and infrastructure provider. Consult your OS or cloud provider's documentation for instructions on creating a swap file or partition.
```

Trying to soften this a bit. The word `Furthermore` doesn't add much value, and the final sentence (This is out of support) is a bit aggresive and unneeded where we are already implying it in the text itself.

What I meant with  my initial review is that our document is not responsible to showing users how to configure swap, but we don't need to elaborate it for such a small topic.

Author: TIL a lot of great things in writing style. Thank you Edu!

> ### Enabling swap [ece-alloc-enabling-swap]

Reviewer: This Enabling swap section needs to be deleted entirely. It's contradictory to the previous sentence, plus the links do NOT tell how to enable swap.

We are already telling users that Swap should be enabled and that to enable swap they need to check their OS guide. That should be enough.
